### PR TITLE
Only display allowed locales with rbac

### PR DESCRIPTION
**Pull request by soupette**
Signed-off-by: soupette <cyril.lpz@gmail.com>

### What does it do?

Everything is said in the title

### CODE REVIEW

File: packages/strapi-plugin-i18n/admin/src/components/LocalePicker/index.js [hunk 1]
```diff
@@ -44,7 +46,14 @@ const LocalePicker = () => {
   const dispatch = useDispatch();
   const pluginOptions = useSelector(selectContentManagerListViewPluginOptions);
   const locales = useSelector(selectI18NLocales);
+  const colllectionTypesRelatedPermissions = useSelector(selectCollectionTypesRelatedPermissions);
   const [{ query }, setQuery] = useQueryParams();
+  const {
```
Comment (mfrachet) [inline packages/strapi-plugin-i18n/admin/src/components/LocalePicker/index.js:51]: Can you create a custom hook somewhere for this? Something like:

```js
const locales = useAllowedLocales()
```

please?

Comment (soupette) [reply packages/strapi-plugin-i18n/admin/src/components/LocalePicker/index.js:51]: yes!

Comment (soupette) [reply packages/strapi-plugin-i18n/admin/src/components/LocalePicker/index.js:47]: I could not create a generic hook for the displayed locales as the logic is a bit different in the edit view. However I have extracted a hook that returns the needed permissions.

Comment (mfrachet) [reply packages/strapi-plugin-i18n/admin/src/components/LocalePicker/index.js:47]: what's different in the edit view?

File: packages/strapi-plugin-i18n/admin/src/components/LocalePicker/index.js [hunk 2]
```diff
@@ -37,20 +40,19 @@ const EllipsisParagraph = styled(Text)`
...
 const LocalePicker = () => {
   const dispatch = useDispatch();
-  const pluginOptions = useSelector(selectContentManagerListViewPluginOptions);
   const locales = useSelector(selectI18NLocales);
   const [{ query }, setQuery] = useQueryParams();
+  const {
+    params: { slug },
+  } = useRouteMatch('/plugins/content-manager/collectionType/:slug');
+  const isFieldLocalized = useHasI18n();
+  const { createPermissions, readPermissions } = useGetContentTypePermissions(slug);
```
Comment (mfrachet) [inline packages/strapi-plugin-i18n/admin/src/components/LocalePicker/index.js:51]:
```suggestion
  const { createPermissions, readPermissions } = useContentTypePermissions(slug);
```

**Review by mfrachet — APPROVED** [review]

**Pull request merged**